### DISCUSSION TITLE
Is there native libraries that work with all java versions?

### DISCUSSION THREAD

**Discussion by tnguy271** [rdkit/rdkit#5651] [Q&A]
I'm using Java 11 now and the native libraries I generated can only work with Java 11, not other versions. Is there anyways to generate native libraries to work with all Java version?

**Comment by greglandrum** [comment]
You haven't provided enough information for us to actually help you here. Please let us know what you've tried and what the problems you encountered were

**Reply by tnguy271** [reply]
Hi Greg, so I'm using Java wrapper for RDKIt for my project. I followed the instruction on rdkit manual to build the rdkit libraries. So I generated the jnilib file with cmake with Java 11 installed on my machine. I was able to load the jnilib file in my project and use the library.   But when I try to share the project with someone else who uses Java 15 they can't load that same native library. So my question was is it possible to generate a jnilib file that works with all Java version? I hope this is clearer
